### PR TITLE
Use CTEST_PARALLEL_LEVEL

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: builder/actions/cmake.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def run(self, env):
             return
 
         ctest = toolchain.ctest_binary()
-        sh.exec(*toolchain.shell_env, ctest,
+        sh.exec(*toolchain.shell_env, ctest, "--parallel", str(os.cpu_count()),
```
Comment (bretambrose) [inline builder/actions/cmake.py:249]: Why?  If this is affecting all repos then that's very concerning.

Comment (TingDaoK) [reply builder/actions/cmake.py:249]: the tests are slow. Hmmm, do we have tests that cannot run in parallel? If so, I guess I'll just change s3 to run in parallel instead

Comment (bretambrose) [reply builder/actions/cmake.py:249]: I worry about crt-cpp where we have mqtt tests that used unique client ids but share topics and cross-talk between them

Comment (bretambrose) [reply builder/actions/cmake.py:249]: Ideally all tests would be truly independent and mutually exclusive, but I don't think we're there yet.

Comment (TingDaoK) [reply builder/actions/cmake.py:249]: Sure, I'll just enable it only for aws-c-s3 then.

**Comment by TingDaoK** [comment]
move to just s3

File: builder/actions/cmake.py (hunk 1 above)
Comment (TingDaoK) [reply builder/actions/cmake.py]: Still needs to update builder for the script to import environment variables

**Review by graebm — APPROVED** [review]

**Pull request merged**